archive/extract_activity.go

```diff
@@ -0,0 +1,177 @@
+package archive
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/google/safeopen"
+	"github.com/mholt/archiver/v4"
+	"go.artefactual.dev/tools/temporal"
+)
+
+const ExtractActivityName = "extract-archive-activity"
+
+var (
+	ErrNotAFile       = errors.New("Not a file")
+	ErrInvalidArchive = errors.New("Invalid archive")
+)
+
+type ExtractActivity struct {
+	cfg Config
+}
+
+func NewExtractActivity(cfg Config) *ExtractActivity {
+	return &ExtractActivity{cfg: cfg}
+}
+
+type ExtractActivityParams struct {
+	// SourcePath is the path of the archive to be extracted
+	SourcePath string
+}
+
+type ExtractActivityResult struct {
+	// DestPath is the path of the extracted archive contents.
+	DestPath string
+}
+
+// Execute extracts the content of SourcePath to a temporary directory, then
+// returns the DestPath of the extracted files.
+//
+// If SourcePath is a directory an ErrNotAFile error is returned. If SourcePath
+// is a file, but not a valid archive, an ErrInvalidArchive error is returned.
+func (a *ExtractActivity) Execute(
+	ctx context.Context, params *ExtractActivityParams,
+) (*ExtractActivityResult, error) {
+	logger := temporal.GetLogger(ctx)
+	logger.V(1).Info("Executing ExtractActivity", "SourcePath", params.SourcePath)
```

Comment:
I think you used V(2) in the other activity.

Reply:
Oops! I think V(1) is better for both as it's the standard practice in https://github.com/artefactual-sdps/enduro.

Reply:
It's also using V(2) below in the switch, where I'm not sure if we should be using Error() instead of Info().

Reply:
@jraddaoui the use of V(2) for ErrNotAFile and ErrInvalidArchive in the switch is intentional, I think these should be a lower priority log message.  

@sevein has argued against using `logger.Error()` in favour of returning an error for most cases, and that makes sense to me.  I think if we want to log errors in Preprocessing we can do that higher up the call stack.  

I'm using `logger.Info()` here and typed errors because I want the flexibility of how to handle these two errors in the caller.  I think in most cases an `ErrNotAFile` should just ignore the error and continue processing with the source directory (it is probably a transfer directory).  Similarly a ErrInvalidArchive is *probably* just a single file transfer, and so processing should continue on that file.  That said, maybe I should leave the logging to the caller in these cases too. :thinking: